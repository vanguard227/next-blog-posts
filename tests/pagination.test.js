import renderer from "react-test-renderer";
import Pagination from "../components/pagination";
import { shallow } from "enzyme";

describe("<Pagination />", () => {
  it("matches snapshot", () => {
    const component = renderer.create(
      <Pagination page={1} setPage={() => {}} total={10} limit={5} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Pagination", () => {
    const wrapper = shallow(<Pagination page={1} setPage={() => {}} total={10} limit={5} />);
    let actual = wrapper.find("label").text();
    expect(actual).toBe("1");

    actual = wrapper.find("button").length;
    expect(actual).toBe(4);
  });

  it("disble first, previous button", () => {
    const wrapper = shallow(<Pagination page={1} setPage={() => {}} total={13} limit={5} />);
    const prevBtn = wrapper.findWhere(
      node => node.props().disabled === true
    )
    let actual = prevBtn.length;
    expect(actual).toBe(2);
  });

  it("disble last, next button", () => {
    const wrapper = shallow(<Pagination page={3} setPage={() => {}} total={13} limit={5} />);
    const prevBtn = wrapper.findWhere(
      node => node.props().disabled === true
    )
    let actual = prevBtn.length;
    expect(actual).toBe(2);
  });
});
