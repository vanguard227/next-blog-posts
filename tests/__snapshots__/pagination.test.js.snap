// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> matches snapshot 1`] = `
<div
  className="sc-bczRLJ dxVBep"
>
  <button
    className="sc-gsnTZi cKArfX"
    disabled={true}
    onClick={[Function]}
  >
    &lt;&lt;
  </button>
  <button
    className="sc-gsnTZi cKArfX"
    disabled={true}
    onClick={[Function]}
  >
    &lt;
  </button>
  <label
    className="sc-dkzDqf eWqzzO"
  >
    1
  </label>
  <button
    className="sc-gsnTZi gwXKYL"
    disabled={false}
    onClick={[Function]}
  >
    &gt;
  </button>
  <button
    className="sc-gsnTZi gwXKYL"
    disabled={false}
    onClick={[Function]}
  >
    &gt;&gt;
  </button>
</div>
`;
